pkgs/by-name/ab/aba/package.nix

```diff
@@ -16,29 +16,28 @@ rustPlatform.buildRustPackage {
     owner = "~onemoresuza";
     repo = "aba";
     rev = version;
-    hash = "sha256-Sz9I1Dw7wmoUPpTBNfbYbehfNO8FK6r/ubofx+FGb04=";
-    domain = "sr.ht";
+    hash = "sha256-2zVQNchL4DFh2v2/kwupJTBSmXiKqlxzUMrP9TbfCMs=";
   };
 
-  cargoSha256 = "sha256-Ihoh+yp12qN74JHvJbEDoYz+eoMwPOQar+yBEy+bqb0=";
+  cargoSha256 = "sha256-YhSzbfcEIJjKWlyYq1lK70qt4f/Z71n7hgaaZ/D/U80=";
 
   nativeBuildInputs = [
     just
     scdoc
   ];
 
   postPatch = ''
-    # Suppress messages of command not found. jq is not needed for the build, but just calls it anyway.
-    sed -i '/[[:space:]]*|[[:space:]]*jq -r/s/jq -r .*/: \\/' ./justfile
-    # Let only nix strip the binary by disabling cargo's `strip = true`, like buildRustPackage does.
+    # Let only nix strip the binary by disabling cargo's `strip = true`, like
+    # buildRustPackage does when not using just's setup hooks.
     sed -i '/strip[[:space:]]*=[[:space:]]*true/s/true/false/' ./Cargo.toml
   '';
 
   preBuild = ''
+    # Used instead of justFlags because of spaces on MANIFEST_OPTS.
```

Comment:
```suggestion
```

Useless comment - this use case is already cited in the documentation.

Reply:
I don't think this is useless. This behavior is surprising at least, even considering that this is in the documentation.

Reply:
So, how should I proceed?

Reply:
> I don't think this is useless. This behavior is surprising at least, even considering that this is in the documentation.

Practically every setup hook contains such `XXXflagsArray` s precisely because sometimes the arguments have spaces. Look at any of them - cmake, meson, scons, waf, even the forgotten bmake.

Why this behavior clearly documented in the chapter about stdenv saying "XXXFlagsArray must be used if the argument contains spaces" and clearly followed by the main setup hooks of build managers is surprising to the point the clear documentation and the huge amount of code is not sufficient to move away the need of this redundand commentary?

Reply:
The only way we wouldn't have a surprising behavior here is if it is fixed, that can be fixed in the future if we eventually switch to structure args.

So IMO keep this comment here to avoid others trying to fix it later and failing is the best choice right now. But I am fine regardless, this is a community in the end and if other people find that this is useless I am also fine.

Reply:
_It's fun when everything is eventually solved in the future_, isn't it?

Regardless, getting rid of Bash in a codebase of 1237 objects and 9752 kilobytes is another whole off-topic issue.

Reply:
Structure args do not need to get rid of Bash, you can manipulate it with e.g.: jq.

But I disgress, back to the topic.

Reply:
So, Should the comment be kept or removed?

Reply:
> Structure args do not need to get rid of Bash, you can manipulate it with e.g.: jq.

I was talking about getting rid of Bash as a programming language instead of mere glue.

>  But I disgress, back to the topic.

Yes.

>  So, Should the comment be kept or removed?

I vote for removing it, since it was not there already **and** it is backed up by the current status quo.

Since we and Okada disagree, you have the Minerva vote.

Reply:
Removed it.